eslint.config.mjs

```diff
@@ -0,0 +1,144 @@
+            },
+          ],
+        },
+      ],
+
+      "react-hooks/exhaustive-deps": "error",
+      "react-hooks/rules-of-hooks": "error",
+
+      "react-native/no-raw-text": "off", // We have so many Text wrapper components... and eslint doesn't know that OnboardingTitleSubtitle.Title is a Text component
+      "react-native/no-color-literals": "warn", // We need to use the colors in theme!
+      "react-native/sort-styles": "off", // Not needed
+      "react-native/no-unused-styles": "off", // Because it's giving warning for styles we use in useStyles hook even though they are used
+      "react-native/no-inline-styles": "warn",
+
+      "react/no-unescaped-entities": "off", // Not needed
+      "react/prop-types": "off", // Disable since we use TypeScript
+      "react/display-name": "off", // Not needed
+      "react/react-in-jsx-scope": "off", // Disable since we use React 18
+      "react/jsx-key": "error",
+      "react/jsx-no-bind": [
+        "warn",
+        {
+          ignoreRefs: true,
+          allowArrowFunctions: false,
+          allowFunctions: false,
+          allowBind: false,
+          ignoreDOMComponents: true,
+        },
+      ],
+
+      "typescript-eslint/no-unused-vars": [
+        "warn",
+        {
+          vars: "all",
+          args: "none",
+          ignoreRestSiblings: true,
+        },
+      ],
+      "typescript-eslint/no-explicit-any": "warn",
+      "typescript-eslint/explicit-function-return-type": "off",
+      "typescript-eslint/consistent-type-definitions": ["warn", "type"],
+
+      "padding-line-between-statements": [
+        "off", // Off for now until we all agree on best practices
+        {
+          blankLine: "always",
+          prev: "multiline-const",
+          next: "multiline-expression",
+        },
+        {
+          blankLine: "always",
+          prev: "multiline-expression",
+          next: "multiline-const",
+        },
+        { blankLine: "always", prev: "multiline-block-like", next: "*" },
+        { blankLine: "always", prev: "*", next: "multiline-block-like" },
+      ],
+
+      "prettier/prettier": "off", // We use prettier manually with linted-staged for example
+
+      "custom-plugin/padding-before-react-hooks": "warn",
+    },
+    settings: {
+      "import/resolver": {
+        typescript: true,
+        node: true,
+      },
+      react: {
+        version: "detect",
+      },
+    },
+  },
+  // Custom logics to enable single function per file for files in utils directory so we can more easily implement tests for them (maybe)
+  // {
+  //   name: "custom/utils-typescript",
+  //   files: ["utils/**/*.ts"],
+  //   languageOptions: {
+  //     parser: parserTypescript,
+  //   },
+  //   plugins: {
+  //     "custom-plugin": customPlugin,
+  //   },
+  //   rules: {
+  //     "custom-plugin/single-function-per-file": "off", // Off for now
+  //   },
+  // },
```

Comment:
Totally agree that this rules makes no sense. I did it first to help with @alexrisch comments on having smaller utils files to make testability better but I agree with internal functions it might get chaotic. I'm also not expert in tests so will let @alexrisch maybe jump in here

Reply:
Agree with @Koleok, testability issues is due to what's getting imported rather than what's getting exported

I'm personally all for this lint rule, but I believe I am likely in the minority, so I'm fine to not have it. I think I would also be the first person to start breaking it too 😆